### PR TITLE
support target specific Info.plist

### LINKED ISSUE #530: EAS displays wrong version numbers in webhook/website for non-primary scheme builds in multi-scheme iOS projects
EAS always displays the version information from `info.plist`, regardless whether that's the right plist for the scheme that's being built.

Reproduction steps:
- Create a bare workflow project
- Create a second target/scheme
- Set unique build numbers on it
- Point to the new scheme in eas.json
- Run a build

Example of such a build: https://expo.dev/accounts/few_io/projects/Unholstered/builds/953bb450-7345-47bd-b925-5816ee5466a8.

You can compare the build versions (1.5, 57) with the values in the IPA.

Bonus points for solving the fact that the _primary_ info.plist doesn't have to be called info.plist, but the tooling assumes it does.

### PR DESCRIPTION
**Pull request by wkozyra95**
# Checklist

- [x] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary.
- [ ] I've tagged the changelog entry with `[EAS BUILD API]` if it's a part of a breaking change to EAS Build API (only possible when updating `@expo/eas-build-job` package).

# Why

Closes #530 

# How

read and write to Info plist based INFOPLIST_FILE

# Test Plan

tested on project plus added some unit tests

### CODE REVIEW

File: packages/eas-cli/src/build/ios/version.ts [hunk 1]
```diff
@@ -31,7 +32,7 @@ export async function bumpVersionAsync({
     return;
   }
   ensureStaticConfigExists(projectDir);
-  const infoPlist = await readInfoPlistAsync(projectDir);
+  const infoPlist = await readInfoPlistAsync(projectDir, {});
```
Comment (wkozyra95) [inline packages/eas-cli/src/build/ios/version.ts:35]: we are removing that either way

File: packages/eas-cli/src/build/ios/__tests__/version-test.ts [hunk 2]
```diff
@@ -30,9 +31,22 @@ beforeEach(() => {
...
 
 describe(evaluateTemplateString, () => {
-  it('evaluates the template string', () => {
+  it('evaluates the template string when value is a number', () => {
     expect(evaluateTemplateString('$(BLAH_BLAH)', { BLAH_BLAH: 123 })).toBe('123');
   });
+  it('evaluates the template string when value is a string', () => {
+    expect(evaluateTemplateString('$(BLAH_BLAH)', { BLAH_BLAH: '123' })).toBe('123');
+  });
+  it('evaluates the template string when template is not the only elment', () => {
```
Comment (dsokal) [inline packages/eas-cli/src/build/ios/__tests__/version-test.ts:40]:
```suggestion
  it('evaluates the template string when template is not the only element', () => {
```

**Review by dsokal — APPROVED** [review]
Fix the lint warning.

File: packages/eas-cli/src/build/ios/version.ts [hunk 3]
```diff
@@ -190,5 +219,12 @@ function ensureStaticConfigExists(projectDir: string): void {
 }
 
 export function evaluateTemplateString(s: string, vars: Record<string, any>): string {
-  return s.replace(/\$\((\w+)\)/g, (match, key) => (vars.hasOwnProperty(key) ? vars[key] : match));
+  return s.replace(/\$\((\w+)\)/g, (match, key) => {
+    if (vars.hasOwnProperty(key)) {
+      const value = String(vars[key]);
+      return value.startsWith('"') ? value.slice(1, -1) : value;
```
Comment (dsokal) [inline packages/eas-cli/src/build/ios/version.ts:225]: Maybe it's worth writing a util for this, I think we have this in few places now.

**Review by dsokal — APPROVED** [review]

**Pull request merged**